File: ios/BlinklinkFeedModule.m
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(BlinklinkFeedModule, RCTEventEmitter)

RCT_EXTERN_METHOD(configure:(NSDictionary *)options)
RCT_EXTERN_METHOD(setUser:(NSString *)ref)
RCT_EXTERN_METHOD(clearUser)
RCT_EXTERN_METHOD(handleUniversalLink:(NSString *)urlString
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: android/src/main/java/com/blinklink/feedreactnative/BlinklinkFeedModule.kt
package com.blinklink.feedreactnative

import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.ReadableMap

/**
 * Android is a no-op in 0.x — the native Android renderer is in progress;
 * the placeholder views confirm the wiring. iOS renders the full experience.
 */
class BlinklinkFeedModule(reactContext: ReactApplicationContext) :
    ReactContextBaseJavaModule(reactContext) {

    override fun getName() = "BlinklinkFeedModule"

    @ReactMethod
    fun configure(options: ReadableMap) = Unit

    @ReactMethod
    fun setUser(ref: String) = Unit

    @ReactMethod
    fun clearUser() = Unit

    @ReactMethod
    fun handleUniversalLink(url: String, promise: Promise) {
        promise.resolve(false)
    }

    // Required stubs for NativeEventEmitter parity.
    @ReactMethod
    fun addListener(eventName: String) = Unit

    @ReactMethod
    fun removeListeners(count: Int) = Unit
}
